### ISSUE TITLE
Annotation formatters not used in th:each loops

### ISSUE THREAD

**Issue by andidev** [thymeleaf/thymeleaf-spring#44]
Thymeleaf seems to ignore Annotation Formatters such as @DateTimeFormat in th:each loops.

E.g.

Html/Thymeleaf

``` html
currentDate = <span th:text="${{object.currentDate}}">Outputs correctly formatted date: 2013-12-16</span>
<br/>

nestedCurrentDate =
<span th:each="nestedObject : ${object.nestedObjectInList}">
    <span th:text="${{nestedObject.nestedCurrentDate}}">Outputs incorrectly unformatted date: 12/16/13 5:00 AM</span>
</span>
```

Controller

``` java
@RequestMapping({"/test"})
public String test(Model model) {
    Object viewModel = new Object();
    model.addAttribute("object", viewModel);
    return "pages/test";
}

public static class Object {
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private DateTime currentDate = DateTime.now();
    private List<NestedObject> nestedObjectInList = Arrays.asList(new NestedObject());
    // ...Getters and Setters
}

public static class NestedObject {
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private DateTime nestedCurrentDate = DateTime.now().minus(1000);
}
```

**Comment by danielfernandez** [comment]
Closing this as a question. Current valid answer is explained at #37.

Note also that ticket #49 has been created in order to study future support for this scenario.

**Comment by danielfernandez** [comment]
Note this has been implemented and will be a part of Thymeleaf 2.1.3. See #49 for more information.
